### ISSUE TITLE
Treat callables consistently

### ISSUE THREAD

**Issue by goodfeli** [tensorflow/cleverhans#761]
Many Attacks support passing in a callable rather than a Model. Currently, they are wildly inconsistent.
Many treat `__call__` as probs (incredibly dangerous, can't believe we're still doing this anywhere) while many treat `__call__` as logits.

I think it would be best to just drop support for all these callables rather than try to change the semantics.
